### PR TITLE
LogIn with Email/Password and Google

### LINKED ISSUE #20: LogIn with Email/Password and Google
## REQUIREMENTS / DESCRIPTION

> Make the LogIn process of Email/Password and Google.

## TASKS
- [x] Implement the funtionality of authentication in the app with React Native logic.
- [x] Handling errors of the authentication (provide feedback to the user).
- [x] Redirect to Home Page when login successfully.
- [x] Use of custom hooks to reduce code duplication.
- [x] Create/Implement a Loder Spinner component to show the process of LogIn and Sign Up.

### PR DESCRIPTION
**Pull request by AlbertoG22**
# Description

What did you implemented/Why did you implemented this:
- Creation and implementation of custom hooks for the form fields, validation of email and password and show/hide password.
- Implementation of the logic to "log in" with the email and google methods.
- Added logic to change the error message as appropriate. Some errors handled are:
  - In sign up if there is already an account created.
  - If the domain in the email is missing.
  - In the login if the email doesn't have an account created.
  - If the password is wrong.
- Creation and implementation of Loader component, when the action is successful it shows a message, if it doesn't shows an error.
- The loader is implemented for both sign up and log in screens, with email and Google authentication methods.

## Screenshots
Texto alternativo
Texto alternativo
Texto alternativo
Texto alternativo
Texto alternativo
Texto alternativo
Texto alternativo
Texto alternativo
Texto alternativo

image

### CODE REVIEW

**Review by Lizzflores — APPROVED** [review]

**Review by landillo — APPROVED** [review]
nice job team!

File: src/styles/AppStyles.js [hunk 1]
```diff
@@ -153,4 +153,29 @@ export const styles = StyleSheet.create({
...
+    flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+    backgroundColor: 'rgba(0, 0, 0, 0.5)',
+  },
+  loadingContent: {
+    // flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+    // backgroundColor: '#343434',
```
Comment (wiltrouble) [inline src/styles/AppStyles.js]: Remove this comment

File: src/styles/AppStyles.js [hunk 2]
```diff
@@ -153,4 +153,29 @@ export const styles = StyleSheet.create({
...
+    // backgroundColor: '#393E51',
```
Comment (wiltrouble) [inline src/styles/AppStyles.js]: Remove this comment

File: src/styles/AppStyles.js [hunk 3]
```diff
@@ -153,4 +153,29 @@ export const styles = StyleSheet.create({
...
+    alignItems: 'center',
+  },
+  overlay: {
+    flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+    backgroundColor: 'rgba(0, 0, 0, 0.5)',
+  },
+  loadingContent: {
+    // flex: 1,
```
Comment (wiltrouble) [inline src/styles/AppStyles.js]: Remove this comment

**Review by wiltrouble — DISMISSED** [review]

**Review by FranciscoSenestrari — APPROVED** [review]

**Pull request merged**